app/models/user.rb

```diff
@@ -0,0 +1,15 @@
+class User < ApplicationRecord
+  has_secure_password
+
+  validates :name, presence: true,
+length: {maximum: Rails.application.config.maximum_name_length}
+
+  validates :email, presence: true, uniqueness: true,
+length: {maximum: Rails.application.config.maximum_email_length},
+format: {with: Rails.application.config.email_regex}
+
+  validates :password, presence: true,
+length: {minimum: Rails.application.config.min_password_length}
+
+  before_save{self.email = email.downcase}
```

Comment:
line code callback này chúng ta nên đưa lên trên đầu file, chứ ko nên thả ở dưới này nhé 
```suggestion
  before_save{selfemail.downcase!}
```
ngoài ra, code như chị suggest dc ko gái